persistence/trees/module_test.go

```diff
@@ -0,0 +1,120 @@
+package trees_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/pokt-network/pocket/internal/testutil"
+	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
+	"github.com/pokt-network/pocket/persistence/trees"
+	"github.com/pokt-network/pocket/runtime"
+	"github.com/pokt-network/pocket/runtime/genesis"
+	"github.com/pokt-network/pocket/runtime/test_artifacts"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+	mockModules "github.com/pokt-network/pocket/shared/modules/mocks"
+)
+
+const (
+	serviceURLFormat = "node%d.consensus:42069"
+)
+
+func TestTreeStore_Create(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockRuntimeMgr := mockModules.NewMockRuntimeMgr(ctrl)
+	mockBus := createMockBus(t, mockRuntimeMgr)
+	genesisStateMock := createMockGenesisState(nil)
+	persistenceMock := preparePersistenceMock(t, mockBus, genesisStateMock)
+
+	mockBus.EXPECT().GetPersistenceModule().Return(persistenceMock).AnyTimes()
+	persistenceMock.EXPECT().GetBus().AnyTimes().Return(mockBus)
+	persistenceMock.EXPECT().NewRWContext(int64(0)).AnyTimes()
+	persistenceMock.EXPECT().GetTxIndexer().AnyTimes()
+
+	treemod, err := trees.Create(mockBus,
```

Comment:
Is there any other validation you can think of after calling create?

E.g. the default root hash
E.g. Calling some of the exposed functions